sydent/util/emailutils.py

```diff
@@ -63,17 +63,22 @@ def sendEmail(
         }
     )
 
-    allSubstitutions = {}
-    for k, v in substitutions.items():
-        allSubstitutions[k] = v
-        allSubstitutions[k + "_forhtml"] = escape(v)
-        allSubstitutions[k + "_forurl"] = urllib.parse.quote(v)
+    # use jinja for rendering if jinja templates are present
+    if templateFile.endswith(".j2"):
+        # We add randomize the multipart boundary to stop user input from
+        # conflicting with it.
+        substitutions["multipart_boundary"] = generateAlphanumericTokenOfLength(32)
+        template = sydent.template_environment.get_template(templateFile)
+        mailString = template.render(substitutions)
+    else:
+        allSubstitutions = {}
+        for k, v in substitutions.items():
+            allSubstitutions[k + "_forhtml"] = escape(v)
+            allSubstitutions[k + "_forurl"] = urllib.parse.quote(v)
+        allSubstitutions["multipart_boundary"] = generateAlphanumericTokenOfLength(32)
+        with open(templateFile) as template_file:
+            mailString = template_file.read() % allSubstitutions
```

Comment:
this seems like we're not preserving the original, unescaped substitutions for old-school templates? (I guess there are no tests for the old-school templates to flag this up).

I suspect this may be wanted:

```suggestion
        allSubstitutions = substitutions.copy()
        for k, v in substitutions.items():
            allSubstitutions[k + "_forhtml"] = escape(v)
            allSubstitutions[k + "_forurl"] = urllib.parse.quote(v)
        allSubstitutions["multipart_boundary"] = generateAlphanumericTokenOfLength(32)
        with open(templateFile) as template_file:
            mailString = template_file.read() % allSubstitutions
```

Reply:
You are right about this but the tests that just got merged into main picked it up and I fixed it

Reply:
I think you're not supposed to use those, but I'm not 100% clear? I guess you might need them for the plaintext version?

Reply:
In the original code we seem to do `allSubstitutions[k] = v`.